src/python/deepgnn/graph_engine/snark/convert.py

```diff
@@ -102,7 +104,7 @@ def convert(self):
         )
 
         d = self.dispatcher
-        dataset = TextFileIterator(
+        dataset = self.iterator_class(
```

Comment:
all iterator classes have to use the same arguments in the constructor? You probably want to pass an object, rather than a class on line 39.